tests/providers/amazon/aws/hooks/test_glue_crawler.py

```diff
@@ -0,0 +1,142 @@
+    'DatabaseName': 'test_db',
+    'Role': mock_role_name,
+    'Targets': {
+        'S3Targets': [
+            {
+                'Path': 's3://test-glue-crawler/foo/',
+                'Exclusions': [
+                    's3://test-glue-crawler/bar/',
+                ],
+                'ConnectionName': 'test-s3-conn',
+            }
+        ],
+        'JdbcTargets': [
+            {
+                'ConnectionName': 'test-jdbc-conn',
+                'Path': 'test_db/test_table>',
+                'Exclusions': [
+                    'string',
+                ],
+            }
+        ],
+        'MongoDBTargets': [
+            {'ConnectionName': 'test-mongo-conn', 'Path': 'test_db/test_collection', 'ScanAll': True}
+        ],
+        'DynamoDBTargets': [{'Path': 'test_db/test_table', 'scanAll': True, 'scanRate': 123.0}],
+        'CatalogTargets': [
+            {
+                'DatabaseName': 'test_glue_db',
+                'Tables': [
+                    'test',
+                ],
+            }
+        ],
+    },
+    'Classifiers': ['test-classifier'],
+    'TablePrefix': 'test',
+    'SchemaChangePolicy': {
+        'UpdateBehavior': 'UPDATE_IN_DATABASE',
+        'DeleteBehavior': 'DEPRECATE_IN_DATABASE',
+    },
+    'RecrawlPolicy': {'RecrawlBehavior': 'CRAWL_EVERYTHING'},
+    'LineageConfiguration': 'ENABLE',
+    'Configuration': """
+    {
+        "Version": 1.0,
+        "CrawlerOutput": {
+            "Partitions": { "AddOrUpdateBehavior": "InheritFromTable" }
+        }
+    }
+    """,
+    'SecurityConfiguration': 'test',
+    'Tags': {'test': 'foo'},
+}
+
+
+class TestAwsGlueCrawlerHook(unittest.TestCase):
+    @cached_property
+    def setUp(self):
+        self.hook = AwsGlueCrawlerHook(aws_conn_id="aws_default", poll_interval=5)
+
+    @unittest.skipIf(mock_iam is None, 'mock_iam package not present')
+    @mock_iam
+    def test_get_iam_execution_role(self):
+        iam_role = self.hook.get_client_type('iam').create_role(
+            Path="/",
+            RoleName=mock_role_name,
+            AssumeRolePolicyDocument=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": {
+                        "Effect": "Allow",
+                        "Principal": {"Service": "glue.amazonaws.com"},
+                        "Action": "sts:AssumeRole",
+                    },
+                }
+            ),
+        )
+        iam_role = self.hook.get_iam_execution_role(role_name=mock_role_name)
+
+        self.assertIsNotNone(iam_role)
+
+    @mock.patch.object(AwsGlueCrawlerHook, "get_iam_execution_role")
+    @mock.patch.object(AwsGlueCrawlerHook, "get_conn")
+    def test_get_or_create_crawler(self, mock_get_conn, mock_get_iam_execution_role):
+        mock_get_iam_execution_role.return_value = mock.MagicMock(Role={'RoleName': mock_role_name})
+
+        mock_glue_crawler = mock_get_conn.return_value.get_crawler()['Crawler']['Name']
```

Comment:
i think what you are trying to do is mock get_crawler to return  some value, and then confirm that get_or_create returns this same value.  but here you are not correctly mocking a return value -- you are just setting `mock_glue_crawler` equal to a mock object `<MagicMock name='get_conn().get_crawler()[41 chars]936'>`